core/node/node_framework/src/service/mod.rs

```diff
@@ -145,30 +148,43 @@ impl ZkStackService {
             .collect();
 
         // Run the tasks until one of them exits.
-        // TODO (QIT-24): wrap every task into a timeout to prevent hanging.
-        let (resolved, idx, remaining) = self
+        let (resolved, resolved_idx, remaining) = self
             .runtime
             .block_on(futures::future::select_all(join_handles));
-        let task_name = tasks[idx].name.clone();
+        let resolved_task_name = tasks[resolved_idx].name.clone();
         let failure = match resolved {
             Ok(Ok(())) => {
-                tracing::info!("Task {task_name} completed");
+                tracing::info!("Task {resolved_task_name} completed");
                 false
             }
             Ok(Err(err)) => {
-                tracing::error!("Task {task_name} exited with an error: {err}");
+                tracing::error!("Task {resolved_task_name} exited with an error: {err}");
                 true
             }
             Err(_) => {
-                tracing::error!("Task {task_name} panicked");
+                tracing::error!("Task {resolved_task_name} panicked");
                 true
             }
         };
 
+        let remaining_tasks_with_timeout: Vec<_> = remaining
+            .into_iter()
+            .map(|task|
+                rt_handle.spawn(async move {
```

Comment:
`task` is already spawned via runtime, you don't really need to wrap it twice.

Reply:
then it panics:
```
thread 'service::tests::test_task_run' panicked at core/node/node_framework/src/service/mod.rs:172:25:
there is no reactor running, must be called from the context of a Tokio 1.x runtime
```